managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerAttributes.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.mledger;
+
+import io.opentelemetry.api.common.Attributes;
+import lombok.Getter;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.opentelemetry.OpenTelemetryAttributes;
+import org.apache.pulsar.opentelemetry.OpenTelemetryAttributes.ManagedLedgerOperationStatus;
+
+@Getter
+public class ManagedLedgerAttributes {
+
+    private final Attributes attributes;
+    private final Attributes attributesOperationSucceed;
+    private final Attributes attributesOperationFailure;
+
+    public ManagedLedgerAttributes(ManagedLedger ml) {
+        var mlName = ml.getName();
+        attributes = Attributes.of(
+                OpenTelemetryAttributes.ML_NAME, mlName,
+                OpenTelemetryAttributes.PULSAR_NAMESPACE, getNamespace(mlName)
+        );
+        attributesOperationSucceed = Attributes.builder()
+                .putAll(attributes)
+                .putAll(ManagedLedgerOperationStatus.SUCCESS.attributes)
+                .build();
+        attributesOperationFailure = Attributes.builder()
+                .putAll(attributes)
+                .putAll(ManagedLedgerOperationStatus.FAILURE.attributes)
+                .build();
+    }
+
+    private static String getNamespace(String mlName) {
+        try {
+            return TopicName.get(TopicName.fromPersistenceNamingEncoding(mlName)).getNamespace();
+        } catch (RuntimeException e) {
+            return null;
```

Comment:
Sometimes, the ML name can be invalid, and `TopicName.get` will throw an exception to signal this. See, for instance, this test: https://github.com/apache/pulsar/blob/774a5d42e8342ee50395cf3626b9e7af27da849e/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerBkEnsemblesTests.java#L403.

Returning `null` instructs OpenTelemetry to not ignore that attribute, so it will not be included. This way, we can have an idea of how many of these names are invalid as well.